openedx/adg/lms/applications/views.py

```diff
@@ -1,3 +1,174 @@
+
+        percentage_of_objectives_completed = '{0:.0%}'.format(
+            user_application_hub.number_of_objectives_completed / self.TOTAL_APPLICATION_OBJECTIVES)
+
+        return render(request, self.template_name, context={
+            'user_application_hub': user_application_hub,
+            'pre_req_courses': pre_req_courses,
+            'percentage': percentage_of_objectives_completed
+        })
+
+    def post(self, request):
+        """
+        Submit user application, send mandrill email according to the Application Confirmation Email format. In the
+        end, it redirects to the application success page.
+
+        Returns:
+            HttpResponse object.
+
+        """
+        user_application_hub = ApplicationHub.objects.get(user=request.user)
+        if not user_application_hub.is_application_submitted:
+            user_application_hub.is_application_submitted = True
+            user_application_hub.submission_date = date.today()
+            user_application_hub.save()
+
+            root_url = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+            course_catalog_url = '{root_url}/courses'.format(root_url=root_url)
+            context = {
+                'course_catalog_url': course_catalog_url
+            }
+            send_mandrill_email(MandrillClient.APPLICATION_SUBMISSION_CONFIRMATION, request.user.email, context)
+
+        return redirect('application_success')
+
+
+class ApplicationSuccessView(RedirectToLoginOrRelevantPageMixin, TemplateView):
+    """
+    View entailing successfully submitted application status of a user.
+
+    """
+
+    template_name = 'adg/lms/applications/success.html'
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a user's application is already submitted or not.
+
+        Returns:
+            Boolean, True or False.
+
+        """
+        try:
+            user_application_hub = ApplicationHub.objects.get(user=self.request.user)
+            return user_application_hub.is_application_submitted \
+                   and user_application_hub.is_prerequisite_courses_passed \
+                   and user_application_hub.is_written_application_completed
+        except ApplicationHub.DoesNotExist:
+            return False
+
+    def handle_no_permission(self):
+        """
+        Redirects on test failure, `test_func()` returns False.
+
+        Returns:
+            HttpResponse object.
+
+        """
+        return HttpResponse(status=400)
+
+    def get_context_data(self, *args, **kwargs):
+        """
+        Send the application submission date for the authenticated user in the context dictionary.
+
+        Returns:
+            dict object.
+
+        """
+        context = super(ApplicationSuccessView, self).get_context_data(*args, **kwargs)
+        user_application_hub = ApplicationHub.objects.get(user=self.request.user)
+        context['submission_date'] = user_application_hub.submission_date
```

Comment:
We could use ` context['submission_date'] = user.application_hub.submission_date` directly without querying separately.